tests/components/lamarzocco/test_number.py

```diff
@@ -126,3 +126,222 @@ async def test_gs3_exclusive_none(
+            ATTR_VALUE: value,
+        },
+        blocking=True,
+    )
+
+    assert len(mock_lamarzocco.configure_prebrew.mock_calls) == 1
+    mock_lamarzocco.configure_prebrew.assert_called_once_with(**kwargs)
+
+
+@pytest.mark.parametrize("device_fixture", [LaMarzoccoModel.GS3_AV])
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+@pytest.mark.parametrize(
+    ("entity_name", "value", "function_name", "kwargs"),
+    [
+        (
+            "prebrew_off_time",
+            6,
+            "configure_prebrew",
+            {"on_time": 3000, "off_time": 6000},
+        ),
+        (
+            "prebrew_on_time",
+            6,
+            "configure_prebrew",
+            {"on_time": 6000, "off_time": 5000},
+        ),
+        ("preinfusion_time", 7, "configure_prebrew", {"off_time": 7000}),
+        ("dose", 6, "set_dose", {"value": 6}),
+    ],
+)
+async def test_pre_brew_infusion_key_numbers(
+    hass: HomeAssistant,
+    mock_lamarzocco: MagicMock,
+    snapshot: SnapshotAssertion,
+    entity_name: str,
+    value: float,
+    function_name: str,
+    kwargs: dict[str, float],
+) -> None:
+    """Test the La Marzocco number sensors for GS3AV model."""
+
+    mock_lamarzocco.current_status["enable_preinfusion"] = True
+
+    serial_number = mock_lamarzocco.serial_number
+
+    func = getattr(mock_lamarzocco, function_name)
+
+    state = hass.states.get(f"number.{serial_number}_{entity_name}")
+    assert state is None
+
+    for key in range(1, KEYS_PER_MODEL[mock_lamarzocco.model_name] + 1):
+        state = hass.states.get(f"number.{serial_number}_{entity_name}_key_{key}")
+        assert state
+        assert state == snapshot(name=f"{serial_number}_{entity_name}_key_{key}-state")
+
+        # service call
+        await hass.services.async_call(
+            NUMBER_DOMAIN,
+            SERVICE_SET_VALUE,
+            {
+                ATTR_ENTITY_ID: f"number.{serial_number}_{entity_name}_key_{key}",
+                ATTR_VALUE: value,
+            },
+            blocking=True,
+        )
+
+        kwargs["key"] = key
+
+        assert len(func.mock_calls) == key
+        func.assert_called_with(**kwargs)
+
+
+@pytest.mark.parametrize("device_fixture", [LaMarzoccoModel.GS3_AV])
+async def test_disabled_entites(
+    hass: HomeAssistant,
+    mock_lamarzocco: MagicMock,
+) -> None:
+    """Test the La Marzocco prebrew/-infusion sensors for GS3AV model."""
+
+    ENTITIES = (
+        "prebrew_off_time",
+        "prebrew_on_time",
+        "preinfusion_time",
+        "set_dose",
+    )
+
+    serial_number = mock_lamarzocco.serial_number
+
+    for entity_name in ENTITIES:
+        for key in range(1, KEYS_PER_MODEL[mock_lamarzocco.model_name] + 1):
+            state = hass.states.get(f"number.{serial_number}_{entity_name}_key_{key}")
+            assert state is None
+
+
+# @pytest.mark.parametrize("device_fixture", [LaMarzoccoModel.LINEA_MICRA])
```

Comment:
Why is this code comment out?

Reply:
I think I was waiting for feedback from you and then just forgot about it